src/test/cc/wfanet/panelmatch/protocol/crypto/commutative_encryption_utility_test.cc

```diff
@@ -0,0 +1,153 @@
+
+namespace wfa::panelmatch {
+namespace {
+using google::protobuf::RepeatedPtrField;
+using ::testing::ContainerEq;
+using ::testing::Not;
+using ::wfanet::IsOk;
+using ::wfanet::IsOkAndHolds;
+using ::wfanet::panelmatch::protocol::crypto::ApplyCommutativeDecryption;
+using ::wfanet::panelmatch::protocol::crypto::ApplyCommutativeEncryption;
+using ::wfanet::panelmatch::protocol::crypto::ReApplyCommutativeEncryption;
+
+using ::wfanet::panelmatch::protocol::protobuf::
+    ApplyCommutativeDecryptionRequest;
+using ::wfanet::panelmatch::protocol::protobuf::
+    ApplyCommutativeDecryptionResponse;
+using ::wfanet::panelmatch::protocol::protobuf::
+    ApplyCommutativeEncryptionRequest;
+using ::wfanet::panelmatch::protocol::protobuf::
+    ApplyCommutativeEncryptionResponse;
+using ::wfanet::panelmatch::protocol::protobuf::
+    ReApplyCommutativeEncryptionRequest;
+using ::wfanet::panelmatch::protocol::protobuf::
+    ReApplyCommutativeEncryptionResponse;
+
+std::vector<std::string> convert_repeatedPtrField_to_vector(
+    RepeatedPtrField<std::string> f) {
+  std::vector<std::string> v(f.begin(), f.end());
+  return v;
+}
+
+void assert_rpf_equal(RepeatedPtrField<std::string> a,
+                      RepeatedPtrField<std::string> b) {
+  ASSERT_THAT(convert_repeatedPtrField_to_vector(a),
+              ContainerEq(convert_repeatedPtrField_to_vector(b)));
+}
+
+void assert_rpf_not_equal(RepeatedPtrField<std::string> a,
+                          RepeatedPtrField<std::string> b) {
+  ASSERT_THAT(convert_repeatedPtrField_to_vector(a),
+              Not(ContainerEq(convert_repeatedPtrField_to_vector(b))));
+}
+
+TEST(PrivateJoinAndComputeTest, EncryptReEncryptDecryptUtility) {
+  std::vector<std::string> plaintexts{"some plaintext0", "some plaintext1",
+                                      "some plaintext2", "some plaintext3",
+                                      "some plaintext4"};
+  RepeatedPtrField<std::string> plaintext_batch(plaintexts.begin(),
+                                                plaintexts.end());
+  std::string random_key_1 = "random-key-1";
+  std::string random_key_2 = "random-key-2";
+
+  ApplyCommutativeEncryptionRequest encrypt_request1;
+  encrypt_request1.set_encryption_key(random_key_1);
+  encrypt_request1.mutable_plaintexts()->CopyFrom(plaintext_batch);
+  auto encrypted_response1 = ApplyCommutativeEncryption(encrypt_request1);
+  ASSERT_THAT(encrypted_response1, IsOk());
+  auto encrypted_texts1 = (*encrypted_response1).encrypted_texts();
+
+  ApplyCommutativeEncryptionRequest encrypt_request2;
+  encrypt_request2.set_encryption_key(random_key_2);
+  encrypt_request2.mutable_plaintexts()->CopyFrom(plaintext_batch);
+  auto encrypted_response2 = ApplyCommutativeEncryption(encrypt_request2);
+  ASSERT_THAT(encrypted_response2, IsOk());
+  auto encrypted_texts2 = (*encrypted_response2).encrypted_texts();
+
+  assert_rpf_not_equal(encrypted_texts2, encrypted_texts1);
```

Comment:
there has to be a cleaner way to do this @efoxepstein but I thought I would just ask you in review. What's the easiest way to compare to values of two RepeatedPtrFields<string>?